test/test_serialization.py

```diff
@@ -1007,6 +1007,501 @@ def _save_load_check(t):
+                          b'data/2PK\x01\x02\x00\x00\x00\x00\x08\x08\x00\x00\x00\x00\x00\x00\xfb\xfd/\x920\x00'
+                          b'\x00\x000\x00\x00\x00\x10\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
+                          b'\x05\x00\x00lstm.save/data/3PK\x01\x02\x00\x00\x00\x00\x08\x08\x00\x00\x00\x00'
+                          b'\x00\x00\xd1\x9egU\x02\x00\x00\x00\x02\x00\x00\x00\x11\x00\x00\x00\x00\x00\x00\x00'
+                          b'\x00\x00\x00\x00\x00\x00\x80\x05\x00\x00lstm.save/versionPK\x06\x06,\x00\x00\x00'
+                          b'\x00\x00\x00\x00\x1e\x03-\x00\x00\x00\x00\x00\x00\x00\x00\x00\x07\x00\x00\x00'
+                          b'\x00\x00\x00\x00\x07\x00\x00\x00\x00\x00\x00\x00\xb8\x01\x00\x00\x00\x00\x00\x00'
+                          b'\xd2\x05\x00\x00\x00\x00\x00\x00PK\x06\x07\x00\x00\x00\x00\x8a\x07\x00\x00\x00\x00'
+                          b'\x00\x00\x01\x00\x00\x00PK\x05\x06\x00\x00\x00\x00\x07\x00\x07\x00\xb8\x01\x00'
+                          b'\x00\xd2\x05\x00\x00\x00\x00')
+
+        buf_le_no_bom = io.BytesIO(data_le_no_bom)
+        buf_le_bom    = io.BytesIO(data_le_bom)
+        buf_be_no_bom = io.BytesIO(data_be_no_bom)
+        buf_be_bom    = io.BytesIO(data_be_bom)
+
+        lstm_le_no_bom = torch.nn.LSTM(3, 3)
+        lstm_le_bom    = torch.nn.LSTM(3, 3)
+        lstm_be_no_bom = torch.nn.LSTM(3, 3)
+        lstm_be_bom    = torch.nn.LSTM(3, 3)
+
+        lstm_le_no_bom.load_state_dict(torch.load(buf_le_no_bom), strict=True)
+        lstm_le_bom.load_state_dict(   torch.load(buf_le_bom),    strict=True)
+        lstm_be_no_bom.load_state_dict(torch.load(buf_be_no_bom), strict=True)
+        lstm_be_bom.load_state_dict(   torch.load(buf_be_bom),    strict=True)
+
+        self.assertEqual(   lstm_le_bom.state_dict(),    lstm_be_bom.state_dict())
+        self.assertNotEqual(lstm_le_no_bom.state_dict(), lstm_be_no_bom.state_dict())
+
+        if (sys.byteorder == 'little'):
+            self.assertEqual(   lstm_le_no_bom.state_dict(), lstm_le_bom.state_dict())
+            self.assertEqual(   lstm_le_no_bom.state_dict(), lstm_be_bom.state_dict())
+            self.assertNotEqual(lstm_be_no_bom.state_dict(), lstm_le_bom.state_dict())
+            self.assertNotEqual(lstm_be_no_bom.state_dict(), lstm_be_bom.state_dict())
+        else:
+            self.assertNotEqual(lstm_le_no_bom.state_dict(), lstm_le_bom.state_dict())
+            self.assertNotEqual(lstm_le_no_bom.state_dict(), lstm_be_bom.state_dict())
+            self.assertEqual(   lstm_be_no_bom.state_dict(), lstm_le_bom.state_dict())
+            self.assertEqual(   lstm_be_no_bom.state_dict(), lstm_be_bom.state_dict())
+
+    def test_serialization_load_bom_complex_data(self):
+        # Generated on LE system using following commands with and without BOM patch:
+        #
+        # import torch
+        # x = torch.randn(2,2, dtype=torch.cfloat)
+        # torch.save(x, 'tensor.le.pt')
```

Comment:
I kind of feel like we should save a list of tensors, of almost all the dtypes we support

Reply:
Sure, will work on that.